Add storage options for Kubernetes agent (#2262)
OctopusDeploy/docs: src/pages/docs/infrastructure/deployment-targets/kubernetes/kubernetes-agent/storage/index.md

Since the Kubernetes agent is built on the Tentacle codebase,  it is necessary to configure shared storage so that the Tentacle Pod can write the files in a place that the Script Pods can read from.

We offer two options for configuring the shared storage - you can use either the default NFS storage or specify a Storage Class name.

## NFS storage

By default, the Kubernetes agent Helm chart will set up an NFS server suitable for use by the agent inside your cluster. The server runs as a `StatefulSet` in the same namespace as the Kubernetes agent, and uses `EmptyDir` storage, as the working files of the agent are not required to be long-lived. 

This NFS server is referenced in the `StorageClass` that the Kubernetes agent and the script pod use. This `StorageClass` will then instruct the `NFS CSI Driver` to mount the server as directed.

This default implementation is made to let you try the Kubernetes agent without worrying about installing a `ReadWriteMany` compatible `StorageClass` yourself. There are  some drawbacks to this approach:

### Reliability

> Since the NFS server runs inside your Kubernetes cluster, upgrades and other cluster operations will interrupt the server. Due to how NFS stores and allows access to the shared data, clients will not be able to recover from a restart cleanly. This results in deployment failure if one happens to be running while the NFS server is restarted.

Reviewer: ```suggestion
Since the NFS server runs inside your Kubernetes cluster, upgrades and other cluster operations can interrupt cause the NFS server to restart. Due to how NFS stores and allows access to shared data, script pods with an NFS volume mount will not be able to recover cleanly from an NFS server restart. This will result in a deployment failure if one happens to be running when the NFS server is restarted.
```

### Privileges

> The NFS server requires `privileged` access when running as a container. This can be disallowed in some clusters, and access to the NFS pod should be kept to a minimum. 

Reviewer: ```suggestion
The NFS server requires `privileged` access when running as a container which may not be permitted depending on cluster configuration.
```

> If you have a use case that can’t tolerate occasional deployment failures, it’s recommended to provide your own `StorageClass` instead of using the default NFS implementation.

Reviewer: Should this have it's own title or be moved to somewhere else?  Maybe under the "Custom StorageClass" title?

Many managed Kubernetes offerings will have offerings that require little effort to use. These will be a “provisioner” (named as such as they “provision” storage for a storage class), which you can then tie to a storage class. Some examples are listed below:

|**Offering**                      |**Provisioner**                    |**Default StorageClass name**       |
|----------------------------------|-----------------------------------|------------------------------------|
|[Azure Kubernetes Service (AKS)](https://learn.microsoft.com/en-us/azure/aks/concepts-storage)    |`file.csi.azure.com`               |`azurefile`                        |
|[Elastic Kubernetes Service (EKS)](https://docs.aws.amazon.com/eks/latest/userguide/storage.html)  |`efs.csi.aws.com`                  |`efs-sc`                            |
|[Google Kubernetes Engine (GKE)](https://cloud.google.com/kubernetes-engine/docs/concepts/storage-overview)    |`filestore.csi.storage.gke.io`     |`standard-rwx`                      |

If you manage your own cluster and don’t have offerings from cloud providers available, there are some in-cluster options you could explore:

> - [Longhorn](https://longhorn.io/)

Author: How do we feel about linking to 3rd parties?

Author: @zentron ☝️
